packages/grid/src/GridRenderer.js

```diff
@@ -89,10 +94,13 @@ class GridRenderer {
     fontWidth = GridRenderer.DEFAULT_FONT_WIDTH
   ) {
     if (width <= 0 || str.length <= 0) {
-      return null;
+      return '';
     }
 
-    const lo = Math.min(Math.floor(width / fontWidth / 2), str.length);
+    const lo = Math.min(
+      Math.max(0, Math.floor(width / fontWidth / 2) - 5),
```

Comment:
Guessing the 5 is a pixel offset - might be better to declare it somewhere for readability.